### ISSUE TITLE
Method call linting again

### ISSUE THREAD

**Issue by CameronBieganek** [julia-vscode/julia-vscode#1225]
The should lint as a method call error, but it doesn't:

<img src="https://user-images.githubusercontent.com/8310743/82351581-5fe88300-99c2-11ea-9659-adbdbefa8c28.png" height="80">

Note that it does produce a method error at the REPL:

```julia
julia> reshape(1:4, "hello", 2)
ERROR: MethodError: no method matching reshape(::UnitRange{Int64}, ::String, ::Int64)
Closest candidates are:
  reshape(::AbstractArray, ::Int64...) at reshapedarray.jl:116
  reshape(::AbstractArray, ::Union{Int64, AbstractUnitRange}...) at reshapedarray.jl:110
  reshape(::AbstractArray, ::Union{Colon, Int64}...) at reshapedarray.jl:117
```

**Comment by CameronBieganek** [comment]
To be honest, this feature seems very experimental and should not be included in the normal extension release at this time. Especially since the setting to disable method-call linting does not work (#1184).

**Comment by CameronBieganek** [comment]
This is making VS Code unusable:

<img src="https://user-images.githubusercontent.com/8310743/82352743-06815380-99c4-11ea-82af-8728d6b33c0f.png" width="450">

**Comment by lewisl** [comment]
100 % agree!

I'd go so far as to say there MUST BE A WAY TO TURN OFF LINTING COMPLETELY.  Slows down the editor.  Finds hundreds of false positives.  I'll just run my code and fix the errors the real compiler finds, thank you very much!

Also, we need a setting to exclude files and directories.  All of the code in my .ipynb_checkpoints directories are linted.  Since most of this stuff is ancient and way beyond the source, it is full of linting errors.  It should be possible to exclude.

**Comment by CameronBieganek** [comment]
> Also, we need a setting to exclude files and directories.

@lewisl Interestingly, there does seem to be a context menu option to disable linting for a file or directory:

<img src="https://user-images.githubusercontent.com/8310743/82609719-adabe980-9b82-11ea-8e0d-7b582b29baf6.png" width="400">

But it doesn't seem to be working at the moment, at least on my Mac.

To be fair, I do find the regular linting to be pretty useful. It's just the method-call error linting that's causing a headache right now.

> I'd go so far as to say there MUST BE A WAY TO TURN OFF LINTING COMPLETELY.

There is a setting to completely turn off linting, `"julia.lint.run": false`, but it doesn't seem to be working for me at the moment. See #1184.

**Comment by davidanthoff** [comment]
Note that we currently have a bug with the configuration settings for linting, but we can hopefully fix that soonish.

**Comment by lewisl** [comment]
Thank you.

- Lewis

On May 23, 2020, at 12:26 PM, David Anthoff <notifications@github.com> wrote:

﻿

Note that we currently have a bug with the configuration settings for linting, but we can hopefully fix that soonish.

**Comment by IljaK91** [comment]
@lewisl There is an option in the extension settings that let you turn off linting completely:

![grafik](https://user-images.githubusercontent.com/13484803/82815730-0688c400-9e9a-11ea-91c0-64aed61a4040.png)

**Comment by lewisl** [comment]
Yup.  But I don’t think it is working right now. They know and are working on it.

**Comment by davidanthoff** [comment]
The language server backend was rewritten in v1.220.0: StaticLint.jl and SymbolServer.jl were folded into JuliaWorkspaces, and completion, hover, signature help, go-to-definition and find-references were reimplemented. This report predates that rewrite, so the code that produced it no longer exists and the behaviour cannot be confirmed against the current engine.

Closing as outdated. If you still see this on a current release, please open a fresh issue with a reproducer and we will look at it against the new engine.

**Comment by lewisl** [comment]
Great!
